client/state/connectionMiddleware/reducer.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	OFFLINE_QUEUE_ADD,
+	OFFLINE_QUEUE_REMOVE
+} from 'state/action-types';
+
+export function actionQueue( state = [], action ) {
+	if ( action.type === OFFLINE_QUEUE_ADD ) {
+		let shouldQueueAction = true;
+
+		if ( action.squash ) {
+			const previousCalls = state.filter( storedAction =>
+				( storedAction.hash === action.hash )
```

Comment:
Break on the first match instead? This can be as simple as peeking at the most recent item in the queue and only adding a new item if the hashes are different.